portal/app/algorithms/battery_load_control.py

```diff
@@ -0,0 +1,52 @@
+import json, datetime
+import numpy as np
+from app.farm_updater import sonnen_api
+
+
+def batt_dispatch():
+    from app.models import FarmDevice, FarmData, DeviceType
+
+    egauge_device = FarmDevice.objects.get(device_uid='46613')
+    sonnen_queryset = FarmDevice.objects.filter(type=DeviceType.SONNEN)
+    farmdata = FarmData.objects.filter(farm_device=egauge_device).order_by('-id')[0:3]
+    test_pen_power = []
+
+    # Getting egauge info
+    for data in farmdata:
+        test_pen_power.append(json.loads(data.device_data)["processed"]["POWER_CIRCUIT1"]+json.loads(data.device_data)["processed"]["POWER_CIRCUIT2"])
+
+    avg_test_pen_power = np.average(np.asarray(test_pen_power))
+
+    # getting batt soc
+    batt1 = FarmData.objects.filter(farm_device=sonnen_queryset[0]).order_by('-id')[0]
+    batt_serial1 = sonnen_queryset[0].device_uid
+    batt2 = FarmData.objects.filter(farm_device=sonnen_queryset[1]).order_by('-id')[0]
+    batt_serial2 = sonnen_queryset[1].device_uid
+
+    soc_batt1 = batt1.device_data['USOC']
+    soc_batt2 = batt2.device_data['USOC']
+
+    hour_day = datetime.datetime.now().hour
+    if hour_day < 7:
+        if soc_batt1 < 10:
+            sonnen_api.SonnenApiInterface().manual_mode_control(serial=batt_serial1, mode='charge', value='2000')
```

Comment:
Why do you keep re-instantiating the SonnenApiInterface? Either just use one instance or make it a static function.
If you make it an instance, than instantiate the individual (unique batteries), then call the manual control on it, or if you make it static, then it would be a bit more similar to what you have. I suggest the former solution as it would make more sense.